editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java

```diff
@@ -308,39 +312,36 @@ private boolean confirmPerspectiveSwitch(IWorkbenchWindow window, IPerspectiveDe
 	 * @param project
 	 */
 	private void openCamelContextFile(IProject project, IProgressMonitor monitor) {
-		if (project != null) {
-			final IFile[] holder = new IFile[1];
-			searchCamelContextXMLFile(project, holder);
-			try {
-				if (holder[0] == null && project.hasNature(JavaCore.NATURE_ID)) {
-					searchCamelContextJavaFile(project, monitor, holder);
-					isJavaEditorToOpen = true;
-				}
-			} catch (CoreException e1) {
-				ProjectTemplatesActivator.pluginLog().logError(e1);
+		if (project == null) return;
```

Comment:
I prefer the previous way of writing it, it helps to notice just with the indentation that the code is executed conditionally